lib/rgit/cli.rb

```diff
@@ -64,4 +51,112 @@ def test(*args)
       puts args
     end
   end
-end
\ No newline at end of file
+end
+
+module Impl
+  class Cli
+    def format_options(option_hash)
+      result = ""
+      option_hash.each { |k, v|
+        key_str = " --#{k.to_s.gsub("_", "-")}"
+        case v # v is truthy in all cases except: nil, false
+        when [true, false].include?(v) # it's a boolean
+          result << key_str
+        when String
+          result << key_str << "=#{v}"
+        when Fixnum
+          result << key_str << "=#{v}"
+        when Hash
+          v.each { |key, val|
+            result << key_str << " #{key}=#{val}"
+          }
+        end
+      }
+      result
+    end
+
+    def lowest_repo_above(start_dir)
+      res_dir = nil
+      start_dir.ascend { |dir|
+        if dir.has_child? ".git"
+          res_dir = dir
+          break
+        end
+      }
+      res_dir
+    end
+
+    def git_command(cmd, opt_str, dir)
+      capture_pty_stdout("git #{cmd} #{opt_str}", dir)
+    end
+
+    # Opens a virtual shell at the specified
+    # dir and runs the given cmd
+    def capture_pty_stdout(cmd, dir)
+      result = ''
+      PTY.spawn("cd #{dir.realpath}; #{cmd}") do |stdout, stdin, pid|
+        begin
+          stdout.each { |line| result += line }
+        rescue Errno::EIO #Done getting output
+          result
+        end
+      end
+      result
+    end
+
+
+    def init_subrepo(parent_dir, subrepo_dir)
+
+    end
+
+    def init_repo(directory, options)
+      # Check if a git repository exists.
+      already_has_git = directory.has_child?(".git")
+      # 1. Initialize a git repository w/ provided options
+      git_command("init", format_options(options), directory)
+
```

Comment:
Line is too long. [86/80]<br>Do not write to stdout. Use Rails' logger if you want to log.